tests/TyConstraintTests.hs

```diff
@@ -206,7 +206,23 @@ recTests = T.testGroup "Record Type Tests"
       [(x0, recTy' [(0, int32Ty), (64, int64Ty)]),
        (x1, int32Ty),
        (x2, recTy' [(0, int32Ty), (64, (ptrTy 64 int32Ty))]),
-       (x3, (ptrTy 64 int32Ty))]
+       (x3, (ptrTy 64 int32Ty))],
+     mkTest "combining record types 1"
```

Comment:
Selfish request: could we add some descriptions of the intended scenarios backing these test cases?

Reply:
Is this the sort of granularity/background you were thinking?

_These tests show how record subtype constraints should be unified during constraint solving. This arises when we want to combine different atomic facts describing offsets from a single memory location. E.g., if we separately learn (1) at `p` there is an `int32` and (2) at `p+32` there is an `int64`, these statements about `p` can be described via the following two atomic subtype constraints: `p <: {0 : int32}` and `p <: {32 : int64}`. Our unification should then combine these constraints on `p` into `p <: {0 : int32, 32 : int64}`._

Reply:
Yes that is great - thanks!